File: todo-frontend/src/app/models/todo.ts
export interface Todo {
  id: string;
  title: string;
  description: string;
  isCompleted: boolean;
  createdAt?: string;
}

export interface CreateTodoRequest {
  title: string;
  description: string;
}

export interface UpdateTodoRequest {
  id: string;
  title: string;
  description: string;
  isCompleted: boolean;
}

File: Todolist2/Todolist2/Domain/Entities/Todo.cs
namespace Todolist2.Domain.Entities;

public class Todo
{
    public Guid Id { get; set; }

    [System.ComponentModel.DataAnnotations.Required]
    [System.ComponentModel.DataAnnotations.MaxLength(100)]
    public string Title { get; set; } = string.Empty;

    [System.ComponentModel.DataAnnotations.MaxLength(500)]
    public string Description { get; set; } = string.Empty;

    public bool IsCompleted { get; set; }

    public DateTime CreatedAt { get; set; }
}
